server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java

```diff
@@ -240,7 +240,15 @@ public boolean isSkipUnavailable(String clusterAlias) {
     }
 
     public Transport.Connection getConnection(String cluster) {
-        return getRemoteClusterConnection(cluster).getConnection();
+        try {
+            return getRemoteClusterConnection(cluster).getConnection();
+        } catch (NoSuchRemoteClusterException e) {
+            if (getRemoteClusterNames().contains(cluster)) {
```

Comment:
I worry that this kind of approach might see inconsistent views of the current set of remote clusters and give confusing results. Can make it so that `org.elasticsearch.transport.RemoteConnectionManager#getAnyRemoteConnection` doesn't throw a `NoSuchRemoteClusterException` instead? That's definitely incorrect, if we have a `RemoteConnectionManager` instance then the remote cluster definitely exists, and conversely that looks to be the only incorrect one.

Reply:
Thanks. I see why you say that now. I have changed it in the last commit(s).